modules/application/application.js

```diff
@@ -0,0 +1,163 @@
+            eventIdentifier.ref.todoList[i].index = count;
+            localArray.push(eventIdentifier.ref.todoList[i]);
+            count += 1;
+          }
+        });
+        eventIdentifier.ref.todoList = localArray;
+        eventIdentifier.ref.onSaveList();
+        ref.updateDom();
+      });
+      button.index = index;
+      button.ref = ref;
+    }, ref);
+
+    const checks = document.querySelectorAll('.action_check');
+    checks.forEach((check, index) => {
+      check.addEventListener('click', (event) => {
+        const refObj = event.currentTarget;
+        if (refObj.ref.todoList[refObj.index].completed) {
+          refObj.ref.todoList[refObj.index].completed = false;
+        } else {
+          refObj.ref.todoList[refObj.index].completed = true;
+        }
+        refObj.ref.onSaveList();
+        ref.updateDom();
+      });
+      check.index = index;
+      check.ref = ref;
+    }, ref);
+
+    const clearBtn = document.querySelector('.custom-btn');
+    clearBtn.addEventListener('click', (event) => {
+      const refObj = event.currentTarget;
+      const filteredTasks = refObj.ref.todoList.filter((item) => {
+        const state = item.completed === false;
+        return state;
+      });
+      refObj.ref.todoList = filteredTasks;
+      refObj.ref.onSaveList();
+      ref.updateDom();
+    });
+    clearBtn.ref = this;
+  }
+
+  onsubmiteventDispatcher = () => {
+    this.addTask.addEventListener('keyup', (event) => {
+      if (event.keyCode !== 13) {
+        return;
+      }
+
+      const refObj = event.currentTarget;
+      const input = refObj.ref.addTask.value;
+      if (!input.replace(/\s/g, '').length || input.length <= 0) {
+        return;
+      }
+
+      if (event.keyCode === 13) {
+        refObj.ref.todoList.push({
+          index: (refObj.ref.todoList.length + 1), description: input, completed: false,
+        });
+        refObj.ref.onSaveList();
+        refObj.ref.addTask.value = '';
+      }
+      refObj.ref.updateDom();
+      event.preventDefault();
+    });
+    this.addTask.ref = this;
+  }
+
+  onediteventDispatcher = () => {
+    const ref = this;
+    const listDesc = document.querySelectorAll('.desc');
+    listDesc.forEach((desc, index) => {
+      desc.addEventListener('keyup', (event) => {
+        if (event.keyCode !== 13) {
+          return;
+        }
+
+        const refObj = event.currentTarget;
+        let input = refObj.value.innerHTML;
+        refObj.value.innerHTML = input.replace('<br>', '');
+        input = refObj.value.innerHTML;
+        if (!input.replace(/\s/g, '').length || input.length <= 0) {
+          return;
+        }
+
+        if (refObj.value.innerHTML !== refObj.ref.todoList[refObj.index].description) {
+          refObj.value.innerHTML = input.replace('<br>', '');
+          refObj.ref.todoList[refObj.index].description = refObj.value.innerHTML;
+          refObj.ref.onSaveList();
+          refObj.value.blur();
```

Comment:
Nice function for storing data to local storage 👍 . I think you should use a more descriptive variable name so the code can be more readable for other developers